src/cirrus/lib2/events.py

```diff
@@ -0,0 +1,353 @@
+import json
+import os
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from functools import wraps
+from logging import Logger, getLogger
+from typing import Any, Callable, Dict, Optional
+
+import boto3
+
+from .enums import StateEnum, WFEventType
+from .eventdb import EventDB
+from .statedb import StateDB
+from .utils import PAYLOAD_ID_REGEX, SNSPublisher
+
+
+@dataclass
+class WorkflowEvent:
+    event_type: WFEventType
+    payload_id: str
+    isotimestamp: str
+    payload_url: Optional[str] = None
+    execution_arn: Optional[str] = None
+    error: Optional[Dict[str, Any]] = None
+
+    def serialize(self):
+        return json.dumps({x: y for (x, y) in vars(self).items() if y})
+
+    def sns_attributes(self):
+
+        attrs = {
+            "event_type": {
+                "DataType": "String",
+                "StringValue": self.event_type,
+            }
+        }
+
+        if match := PAYLOAD_ID_REGEX.match(self.payload_id):
+            attrs["workflow"] = {
+                "DataType": "String",
+                "StringValue": match.group("workflow"),
+            }
+            attrs["collections"] = {
+                "DataType": "String",
+                "StringValue": match.group("collections"),
+            }
```

Comment:
Can we add just one more attribute for compatibility with the existing failed/invalid message format?

```
if self.error is not None:
    attrs["error"] = {"DataType": "String", "StringValue": error}
```